extensions/VisualEditor/ApiVisualEditor.php

```diff
@@ -367,7 +385,12 @@ public function execute() {
 		wfDebugLog( 'visualeditor', "called on '$page' with paction: '{$params['paction']}'" );
 		switch ( $params['paction'] ) {
 			case 'parse':
-				$parsed = $this->getHTML( $page, $parserParams );
+				if ( isset( $params['preload'] )) {
+					$parsed = $this->getPreloadHTML( $params );
```

Comment:
If $parsed is false it will later return "Error contacting the Parsoid server" (https://github.com/Wikia/app/blob/INT-304/extensions/VisualEditor/ApiVisualEditor.php#L541) but this might be not the case (getPreloadHTML returns false for different reason).
Also, I think you should be passing to getPreloadHTML already just a Title (object) or title (string) of the article you want to retrieve.